### PR TITLE
maint: Add dependency grouping

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Adds dependency grouping to the dependabot so we have to deal with less dependabot PRs.

### CODE REVIEW

**Comment by AlejandroFernandezLuces** [comment]
@ninad-kamat Please take a look and let me know if this setup would work for you.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -12,7 +12,23 @@ updates:
     ignore:
       - dependency-name: "vtk"
       - dependency-name: "grpcio"
-
+    groups:
+      general-dependencies:
+        patterns:
+          - "*"
+        exclude-patterns:
+          - "sphinx*"
```
Comment (RobPasMue) [inline .github/dependabot.yml:20]:
```suggestion
          - "*sphinx*"
          - "numpydoc"
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -12,7 +12,23 @@ updates:
...
+          - "pyvista*"
+          - "pytest*"
+      test-dependencies:
+        patterns:
+          -"pytest*"
+      doc-dependencies:
+        patterns:
+          -"sphinx*"
```
Comment (RobPasMue) [inline .github/dependabot.yml:28]:
```suggestion
          - "*sphinx*"
          - "numpydoc"
```

**Review by RobPasMue — APPROVED** [review]
Left some suggestions. Otherwise LGTM

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**